### PR TITLE
Ripping condoms

### PR DESCRIPTION
**Pull request by picobyte**
This (code-only) pull request should enable ripping condoms. The clothingtype physicalattribute now becomes a float. for condoms it is the fraction chance that a condom doesn't rip during penetration. If ripped, a condom receives the (new) itemEffectType BROKEN_CONDOM that causes SexActioniInterface.ignoreCondom() and cum in/on areas during orgasm; function applyGenericDescriptionsAndEffects(). Some functions were also cleaned up a bit.

This has been compile tested only for now. I almost never use discord, but if I do it's the same name.

### CODE REVIEW

**Comment by picobyte** [comment]
```
	GENERAL_INTEGRITY("integrity",
			"Modifies the integrity of an object.",
			"integrity change",
			"integrity_change",
			Colour.BASE_ROSE,
			Rarity.COMMON),

	public static AbstractItemEffectType INTEGRITY = new AbstractItemEffectType(Util.newArrayListOfValues(
			"Is fragile."), Colour.BASE_ROSE) {

		@Override
		public List<String> getEffectsDescription(TFModifier primaryModifier, TFModifier secondaryModifier,
				TFPotency potency, int limit, GameCharacter user, GameCharacter target) {

			if(primaryModifier!=null && primaryModifier == TFModifier.GENERAL_INTEGRITY) {
				switch(potency) {
				case MAJOR_DRAIN: return Util.newArrayListOfValues("Severily compromises integrity.");
				case DRAIN:       return Util.newArrayListOfValues("Deteriorates integrity.");
				case MINOR_DRAIN: return Util.newArrayListOfValues("Affects integrity a bit.");
				case MINOR_BOOST: return Util.newArrayListOfValues("Mends it somewhat.");
				case BOOST:       return Util.newArrayListOfValues("Repairs it.");
				case MAJOR_BOOST: return Util.newArrayListOfValues("Restores it.");
				}
			}
			return null; // currently impossible.
		}

		@Override
		public String applyEffect(TFModifier primaryModifier, TFModifier secondaryModifier, TFPotency potency,
				int limit, GameCharacter user, GameCharacter target, ItemEffectTimer timer) {

			if (limit != 0) {
				if (secondaryModifier == TFModifier.CRITICAL_CHANCE) {
					if(primaryModifier!=null && primaryModifier == TFModifier.GENERAL_INTEGRITY) {
						switch(potency) {
						case MAJOR_DRAIN:
							if (Util.random.nextInt(limit + 1) > limit / 2)
								limit = 0;
							break;
						case DRAIN:
							if (Util.random.nextInt(limit + 1) > limit / 4)
								limit = 0;
							break;
						case MINOR_DRAIN:
							if (Util.random.nextInt(limit + 1) > limit / 8)
								limit = 0;
							break;
						default:
						}
					}
				}
			}
			if (limit == 0)
				return "<p>It is broken beyond repair</p>";

			return "<p>It still seems ok.</p>";

		}
	};
```

**Comment by picobyte** [comment]
I opted not to propagate the ItemTag/ItemEffect madness, but to give the physicalResistance
another function. The usage of its high bits for this purpose ensures this does not interfere with the current functionality. 

Leave physicalResistance an int. Use high bits to store a chance of breaking.

Do condom rip test in ignoreCondom() renamed to testCondomRipped()
since this is called after it is established the passed gamecharacter is
a condom wearer, there is no reason to null test for penis/condom here.
Now also includes cum production in chance whether condom breaks.
Also change the BROKEN_CONDOM itemeffect to GENERIC_BROKEN so it can be used for
other purposes as well.

**Comment by picobyte** [comment]
I changed it again, reverted the changes to physicalResistance, implemented it with status effects. Added mineral oil secretion (as a female secretion) that deteriorates latex[1].
[1] https://www.sciencedirect.com/science/article/pii/0010782489900188

File: src/com/lilithsthrone/game/inventory/enchanting/ItemEffectType.java [hunk 1]
```diff
@@ -90,6 +90,53 @@ public String applyEffect(TFModifier primaryModifier, TFModifier secondaryModifi
...
+				return Util.newArrayListOfValues("It is [style.boldMinorGood(reliable)].");
+			case MINOR_DRAIN:
+				return Util.newArrayListOfValues("It is [style.boldMinorBad(a little weak)].");
+			default:
+				return null;
+			}
+		}
+		@Override
+		public String applyEffect(TFModifier primaryModifier, TFModifier secondaryModifier, TFPotency potency, int limit, GameCharacter user, GameCharacter target, ItemEffectTimer timer) {
+			return genericAttributeEffect(ResourceRestoration.MANA, primaryModifier, secondaryModifier, potency, limit, user, target);
```
Comment (CognitiveMist) [inline src/com/lilithsthrone/game/inventory/enchanting/ItemEffectType.java]: Should this just be `return "";`?

File: src/com/lilithsthrone/game/sex/sexActions/SexActionInterface.java [hunk 2]
```diff
@@ -913,7 +921,57 @@ default boolean performPhysicallyBlockedCheck(SexAreaInterface sArea, GameCharac
...
 	
-	public default boolean ignoreCondom(GameCharacter condomWearer) {
+	public default boolean testCondomRipped(GameCharacter condomWearer, GameCharacter cumTarget) {
+		AbstractClothing condom = condomWearer.getClothingInSlot(ClothingType.PENIS_CONDOM.getSlot());
+		int max = 100;
+		for (ItemEffect eff: condom.getEffects()) {
+			AbstractItemEffectType effType = eff.getItemEffectType();
+			if (effType.equals(ItemEffectType.GENERIC_BROKEN)) {
+				return true;
+			} else if (effType.equals(ItemEffectType.GENERIC_DURABILITY)) {
```
[the remark this answers is not in the document]
Comment (picobyte) [reply src/com/lilithsthrone/game/sex/sexActions/SexActionInterface.java]: If you think something is missng for this to work let me know, if the question was what the intetion was:
The idea was: you can have a normal condom, then there's 1% chance during orgasm it breaks plus 1% per every 10ml of cum produced. Then there's the option a condom gets a GENERIC_DURABILITY modifier, which can have 6 different potencies.
With a BOOST, for instance, the chance is 0 to 200, so there's 0.5% chance during orgasm it breaks plus 0.25% per every 10ml of cum produced. With a DRAIN it's 2% change + 4% per 10ml cum. The vagina also can have a BOOST-like effect, if slimy, and DRAIN-like effect if mineral oily.

File: src/com/lilithsthrone/game/inventory/enchanting/ItemEffectType.java (hunk 1 above)
Comment (picobyte) [reply src/com/lilithsthrone/game/inventory/enchanting/ItemEffectType.java]: Good catch, I stole that, wasn't sure what to return but MANA is obviously out of place.

File: src/com/lilithsthrone/game/sex/sexActions/SexActionInterface.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (CognitiveMist) [reply src/com/lilithsthrone/game/sex/sexActions/SexActionInterface.java]: Oh, I see. I hadn't noticed that the logic still works as expected without needing a special durability value.

**Comment by Innoxia** [comment]
Thank you for this! Sorry it took so long to merge! ^^

I'll run over the condom feature list tomorrow and use your code to implement as many of them as possible.

**Pull request merged**